src/cogs/admin_commands.py

```diff
@@ -0,0 +1,96 @@
+        users_who_started_survey = db.get_all_user_ids_from_survey_progress(
+            WELCOME_SURVEY_ID
+        )
+        for member in get_server(self.bot).members:
+            if member != self.bot.user and member.id not in users_who_started_survey:
+                if discord.utils.get(member.roles, id=ADMIN_ROLE_ID) is None:
+                    await welcome_member(self.bot, member)
+                    sent_to.append(member.display_name)
+        await ctx.channel.send(f"Sent to {sent_to}")
+
+    @commands.command(name="ping-unanswered-survey")
+    @commands.has_role(ADMIN_ROLE_ID)
+    async def ping_users_with_unanswered_questions(self, ctx):
+        """
+        Only member with admin role can run this command
+        Command pings all users who have started welcome_survey (before interval) but haven't answered any questions
+        """
+        pinged = []
+        logging.info(
+            f"Executing ping-unanswered-survey. OLDER_THAN is set to {PING_UNANSWERED_SURVEY_OLDER_THAN}"
+        )
+        users_from = db.get_all_in_progress_users_with_channel_from_survey_progress_created_older_than(
+            WELCOME_SURVEY_ID, PING_UNANSWERED_SURVEY_OLDER_THAN
+        )
+        if len(users_from) > 0:
+            for user, user_channel in users_from.items():
+                channel = self.bot.get_channel(user_channel)
+                await channel.send(PING_UNANSWERED_SURVEY_MESSAGE)
+                pinged.append(user)
+        await ctx.channel.send(f"Pinged {len(pinged)} users: {pinged}")
+
+    @commands.command(name="delete-finished-surveys-channels")
+    @commands.has_role(ADMIN_ROLE_ID)
+    async def delete_finished_surveys_channels(self, ctx):
+        """
+        Only member with admin role can run this command
+        Command deletes channel of surveys which have been completed before DELETE_SURVEYS_OLDER_THAN
+        """
+        deleted_channels = []
+        logging.info(
+            f"Executing delete_finished_surveys_channels command. OLDER_THAN is set to {DELETE_FINISHED_SURVEYS_OLDER_THAN}"
+        )
+        finished_surveys_channel_ids = db.get_completed_survey_channel_ids_older_than(
+            DELETE_FINISHED_SURVEYS_OLDER_THAN
+        )
+        for channel_id in finished_surveys_channel_ids:
+            channel = self.bot.get_channel(channel_id)
+            if channel is None:
+                logging.info(
+                    f"Channel ({channel_id}) not found on server but is in finished_surveys_channel_ids."
+                )
+                continue
+
+            logging.info(f"Deleting channel ({channel_id}).")
+            await channel.delete()
+            db.set_user_survey_progress_status_to_channel_deleted(channel_id)
+            deleted_channels.append(channel_id)
+
+        await ctx.channel.send(f"Deleted {len(deleted_channels)} channels.")
+
+    async def cog_command_error(self, ctx, error):
+        logging.error(
```

Comment:
I would add something like this:
```suggestion
        if isinstance(error, commands.MissingRole):
            logging.error(f"Unauthorized member {ctx.author} called {ctx.command} command.")
        else:
            logging.error(f"Command {ctx.command} error: {error}.")
```

Reply:
@xjahic The error handler in `main.py` also reacts to this error and logs the whole stack trace. We should address this at some point. I've created a task for this.

Reply:
The error handler in `main.py` could be removed all together. There are no commands outisde of `admin_commands.py` and each new cog could be implemented with its own error handler, so that it can have more specific error handling.